nilearn/plotting/surf_plotting.py

```diff
@@ -503,35 +503,25 @@ def plot_surf_roi(surf_mesh, roi_map, bg_map=None,
     nilearn.plotting.plot_surf: For brain surface visualization.
     """
 
-    v, _ = load_surf_mesh(surf_mesh)
-
-    # if roi_map is a list of arrays with indices for different rois
-    if isinstance(roi_map, list):
-        roi_list = roi_map[:]
-        roi_map = np.zeros(v.shape[0])
-        idx = 1
-        for arr in roi_list:
-            roi_map[arr] = idx
-            idx += 1
-
-    elif isinstance(roi_map, np.ndarray):
-        # if roi_map is an array with values for all surface nodes
-        roi_data = load_surf_data(roi_map)
-        # or a single array with indices for a single roi
-        if roi_data.shape[0] != v.shape[0]:
-            roi_map = np.zeros(v.shape[0], dtype=int)
-            roi_map[roi_data] = 1
+    # preload roi and mesh to determine vmin, vmax and give more useful error
+    # messages in case of wrong inputs
 
-    else:
-        raise ValueError('Invalid input for roi_map. Input can be a file '
-                         '(valid formats are .gii, .mgz, .nii, '
-                         '.nii.gz, or Freesurfer specific files such as '
-                         '.annot or .label), or a Numpy array containing a '
-                         'value for each vertex, or a list of Numpy arrays, '
-                         'one array per ROI which contains indices of all '
-                         'vertices included in that ROI')
-    vmin, vmax = np.min(roi_map), 1 + np.max(roi_map)
-    display = plot_surf(surf_mesh, surf_map=roi_map, bg_map=bg_map,
+    roi = load_surf_data(roi_map)
+    vmin, vmax = np.min(roi), 1 + np.max(roi)
+
+    mesh = load_surf_mesh(surf_mesh)
+
+    if len(roi.shape) is not 1:
+        raise ValueError('roi_map can only have one dimension but has'
```

Comment:
I think that you are missing a space at the end of the string

Reply:
Done